Introduce attribute reflection extended attributes

lukewarlow opened the pull request:
A number of extended attributes are added which trigger IDL attribute reflection:

- [Reflect] and [Reflect=string] cover basic content attribute reflection cases.
- [ReflectSetter] and [ReflectSetter=string] covers content attribute reflection only for setters.
- [ReflectURL] and [ReflectURL=string] covers reflection of USV attributes, which represent URLs.

- [ReflectNonNegative] and [ReflectNonNegative=string] covers reflection of long attributes that should be limited to only non-negative numbers.

- [ReflectPositive] and [ReflectPositive=string] covers reflection of double and unsigned long attributes which should be limited to positive numbers.

- [ReflectPositiveWithFallback] and [ReflectPositiveWithFallback=string] covers reflection of double and unsigned long attributes which should be limited to positive numbers with fallback.

Additionally:
- [ReflectRange=(integer, integer)] can be used alongside others on unsigned long attributes that should be clamped to a range.

- [ReflectDefault=number] can be used alongside others on double, long and unsigned long attributes that should be reflected with a default value.

Depends on: https://github.com/whatwg/webidl/pull/1503

<!--
Thank you for contributing to the HTML Standard! Please describe the change you are making and complete the checklist below if your change is not editorial.

When you submit this PR, and each time you edit this comment (including checking a checkbox through the UI!), PR Preview will run and update it. As such make any edits in one go and only after PR Preview has run.

If you think your PR is ready to land, please double-check that the build is passing and the checklist is complete before pinging.
-->

- [x] At least two implementers are interested (and none opposed):
   * N/A, because it's editorial
- [x] [Tests](https://github.com/web-platform-tests/wpt) are written and can be reviewed and commented upon at:
   * N/A
- [x] [Implementation bugs](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) are filed:
   * Chromium: https://issues.chromium.org/u/1/issues/434737815
   * Gecko: https://bugzilla.mozilla.org/show_bug.cgi?id=1979778
   * WebKit: https://bugs.webkit.org/show_bug.cgi?id=296589
   * Servo: https://github.com/servo/servo/issues/38317
   * Ladybird: https://github.com/LadybirdBrowser/ladybird/issues/5631
   * JSDOM: https://github.com/jsdom/jsdom/issues/3913
- [x] Corresponding [HTML AAM](https://w3c.github.io/html-aam/) & [ARIA in HTML](https://w3c.github.io/html-aria/) issues & PRs: N/A
- [x] [MDN issue](https://github.com/whatwg/meta/blob/main/MAINTAINERS.md#handling-pull-requests) is filed: N/A
- [x] The top of this comment includes a [clear commit message](https://github.com/whatwg/meta/blob/main/COMMITTING.md) to use. <!-- If you created this PR from a single commit, Github copied its message. Otherwise, you need to add a commit message yourself. -->

See #3238 

(See [WHATWG Working Mode: Changes](https://whatwg.org/working-mode#changes) for more details.)


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://whatpr.org/html/11450/common-dom-interfaces.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/common-dom-interfaces.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/common-dom-interfaces.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/dom.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/dom.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/dom.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/edits.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/edits.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/edits.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/embedded-content-other.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/embedded-content-other.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/embedded-content-other.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/embedded-content.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/embedded-content.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/embedded-content.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/form-control-infrastructure.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/form-control-infrastructure.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/form-control-infrastructure.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/form-elements.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/form-elements.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/form-elements.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/forms.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/forms.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/forms.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/grouping-content.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/grouping-content.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/grouping-content.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/iframe-embed-object.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/iframe-embed-object.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/iframe-embed-object.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/image-maps.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/image-maps.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/image-maps.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/index.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/index.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/index.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/input.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/input.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/input.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/interaction.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/interaction.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/interaction.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/interactive-elements.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/interactive-elements.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/interactive-elements.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/links.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/links.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/links.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/media.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/media.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/media.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/obsolete.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/obsolete.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/obsolete.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/popover.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/popover.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/popover.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/scripting.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/scripting.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/scripting.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/semantics.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/semantics.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/semantics.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/tables.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/tables.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/tables.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )
<a href="https://whatpr.org/html/11450/text-level-semantics.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">/text-level-semantics.html</a>  ( <a href="https://whatpr.org/html/11450/baec061...cb35b66/text-level-semantics.html" title="Last updated on Jul 30, 2025, 12:16 PM UTC (cb35b66)">diff</a> )

domenic commented:
I'm having a hard time reviewing this because, despite the bold text in the beginning, it seems to mix up the enumerated attribute handling into the same PR as the non-enumerated attribute work.

lukewarlow commented:
I've dropped the second commit from this PR so it should be reviewable now.

annevk commented on source:
```diff
@@ -8257,7 +8257,8 @@ a.setAttribute('href', 'https://example.com/'); // change the content attribute
      <li>
-      <p>If <var>attributeDefinition</var> indicates it contains a <span>URL</span>:</p>
+      <p>If <var>attributeDefinition</var> indicates it contains a <span>URL</span> or the
+      <span>reflected IDL attribute</span> is <span>treated as a URL</span>:</p>
```
Why do we need both? I was hoping that the IDL attributes would obviate the need for _attributeDefinition_.

lukewarlow replied:
Ah good catch I think I left that in case there were any that needed manual handling but [ReflectURL] I think covers all such cases.

annevk reviewed (commented):
Would it be possible to start with `[ReflectURL]` and take them in turns? I don't really know what @domenic would prefer so wait for him to say something, but I would find it easier if we did them incrementally.

lukewarlow commented:
> Would it be possible to start with [ReflectURL] and take them in turns? I don't really know what @domenic would prefer so wait for him to say something, but I would find it easier if we did them incrementally.

To clarify you mean add a PR that does just `[ReflectURL]` and then a follow up that does *just* `[Reflect]`?

I'm happy to split this however reviewers think is best.

domenic reviewed (commented):
Overall this is quite impressive. The comprehensive updates to the prose and IDL are wonderful. I especially like the touch of adding the `data-x`s to the string arguments so you link to the content attribute definitions.

In terms of splitting the PR, I think it would be reasonable to either do just [Reflect], or do [Reflect] and [ReflectURL], or do all of these at once. I don't think it will change the complexity of the review too much, and I don't feel strongly which path we take.

This specification text direction, which builds on top of and preserves the existing prose-based infrastructure, is different than what I was expecting. I was picturing that we would rebuild the specification to be centered around the extended attributes, getting rid of concepts like "reflected target", "reflected IDL attribute", "reflected content attribute", "treated as a URL", "limited to only positive numbers", etc. Instead we would do something like:

> IDL attributes annotated with the [Reflect] content attribute have the following getter steps: [...] and the following setter steps: [...]

and then everything there would be phrased in terms of the IDL concepts, like the extended attribute arguments, the IDL attribute type, `<span>this</span>`, or the presence of other auxiliary extended attributes.

I think my vision is cleaner, as it has fewer layers of abstraction. It also prevents people from attempting to do reflection in prose, which seems like a good idea. And it's closer to how one might implement these. However, it's certainly less incremental. What do you think?

lukewarlow commented:
> This specification text direction, which builds on top of and preserves the existing prose-based infrastructure, is different than what I was expecting. I was picturing that we would rebuild the specification to be centered around the extended attributes, getting rid of concepts like "reflected target", "reflected IDL attribute", "reflected content attribute", "treated as a URL", "limited to only positive numbers", etc. Instead we would do something like:

> IDL attributes annotated with the [Reflect] content attribute have the following getter steps: [...] and the following setter steps: [...]

> and then everything there would be phrased in terms of the IDL concepts, like the extended attribute arguments, the IDL attribute type, <span>this</span>, or the presence of other auxiliary extended attributes.

> I think my vision is cleaner, as it has fewer layers of abstraction. It also prevents people from attempting to do reflection in prose, which seems like a good idea. And it's closer to how one might implement these. However, it's certainly less incremental. What do you think?

So my thought process was to build this in a way that was as incremental as possible. Which is why it's currently in the form I've used.

I don't know that we'll ever be able to fully get rid of spec prose for reflection, but it might be that the end situation has few enough special cases that their prose should just be rewritten to not mention reflection and then we could drop many of these existing concepts.

My other hesitation along the same lines is I don't know what other specs use these algorithms. I didn't want to accidentally break URL reflection from another spec by removing the prose concept (though I'll admit it's rather vague at the moment so it's ideal anyway).

I think my preference would be to write them in terms of the existing prose initially and then once we've moved everything across (including looking at other specs). We can possibly rewrite how they're defined to drop the old approach. It would be purely editorial (I guess normative because we're removing existing algorithms?).

That way we don't break existing stuff and avoid duplication at the same time?

One other question is are we aiming to go for 100% coverage in IDL. Or is it okay if we end up with some still defined manually.

There's a number of odd cases that aren't currently possible to do in the IDL proposal. 

- [Document.dir]((https://html.spec.whatwg.org/multipage/dom.html#dom-document-dir)) reflects the content attribute of the html element 

- [form.action](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fs-action) and form.formAction's missing and empty value default is to return the document's URL instead (If we had a ReflectSetter IDL attribute that would aleviate the simple setter side, but the getter will always be custom).

- [.autocomplete](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fe-autocomplete) returns IDL-exposed autofill value internal state (ditto about ReflectSetter)

~~- [canvas.width](https://html.spec.whatwg.org/multipage/canvas.html#dom-canvas-width) and canvas.height have potentially dynamic default values though the spec prose isn't clear on what it actually means.~~ I've just realised that these can be done with [ReflectNonNegative, ReflectDefault=300] (or 150 for height)

- [.tabIndex](https://html.spec.whatwg.org/multipage/interaction.html#dom-tabindex) tabIndex has a defaultValue that's dependent on element itself. It's currently defined on a mixin so this wouldn't work. It's even based on where the summary element is in the DOM so I don't think this could be handled by IDL. (Potentially we could use [Reflect] and then just override the defaultValue using spec prose, but that would require we keep the default value concept in prose.

- Various document IDL attributes reflect content attributes from the body element (or use an empty string if it's a frameset) so again that couldn't be defined in IDL.

Worth noting that stuff such as [button.command](https://html.spec.whatwg.org/multipage/form-elements.html#dom-button-command) are in a similar boat where maybe the setter could have used reflection but the getter is too complex for that. If we added [ReflectSetter] there's cases like this that could be handled.

annevk commented on source:
```diff
@@ -11923,16 +12008,16 @@ interface <dfn interface>HTMLElement</dfn> : <span>Element</span> {
   // metadata attributes
-  [<span>CEReactions</span>] attribute DOMString <span data-x="dom-title">title</span>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>] attribute DOMString <dfn attribute for="HTMLElement" data-x="dom-title">title</dfn>;
   [<span>CEReactions</span>] attribute DOMString <span data-x="dom-lang">lang</span>;
```
Yes, that's correct.

lukewarlow commented on source:
```diff
@@ -64888,9 +64577,9 @@ interface <dfn interface>HTMLTemplateElement</dfn> : <span>HTMLElement</span> {
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>="<code data-x="attr-template-shadowrootdelegatesfocus">shadowrootdelegatesfocus</code>"] attribute boolean <dfn attribute for="HTMLTemplateElement" data-x="dom-template-shadowrootdelegatesfocus">shadowRootDelegatesFocus</dfn>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>="<code data-x="attr-template-shadowrootclonable">shadowrootclonable</code>"] attribute boolean <dfn attribute for="HTMLTemplateElement" data-x="dom-template-shadowrootclonable">shadowRootClonable</dfn>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>="<code data-x="attr-template-shadowrootserializable">shadowrootserializable</code>"] attribute boolean <dfn attribute for="HTMLTemplateElement" data-x="dom-template-shadowRootSerializable">shadowRootSerializable</dfn>;
   [<span>CEReactions</span>] attribute DOMString <span data-x="dom-template-shadowRootCustomElementRegistry">shadowRootCustomElementRegistry</span>;
```
TODO: This can use [Reflect=] too, the DOMString should just reflect "true" or "false" from the boolean attribute as I understand it.

lukewarlow commented:
Hopefully some of these changes improve the clarity of the definitions. I've also added a [ReflectSetter] as it turns out WebKit already has it and it cuts down on a number of basic setters.

domenic commented on source:
```diff
@@ -8804,6 +8800,103 @@ a.setAttribute('href', 'https://example.com/'); // change the content attribute
+  <p>The <code data-x="xattr-ReflectDefault">[ReflectDefault]</code> <span>extended attribute</span>
+  must only be used on attributes with a type of <code data-x="idl-double">double</code>, <code
+  data-x="idl-long">long</code>, or <code data-x="idl-unsigned-long">unsigned long</code>. When used
+  on an attribute of type <code data-x="idl-double">double</code>, it must take a decimal, otherwise
```
```suggestion
  on an attribute of type <code data-x="idl-double">double</code>, it must take a decimal; otherwise
```

domenic commented on source:
```diff
@@ -8804,6 +8800,103 @@ a.setAttribute('href', 'https://example.com/'); // change the content attribute
+
+  <p>IDL attributes with the <code data-x="xattr-ReflectDefault">[ReflectDefault]</code>
+  <span>extended attribute</span> have a <span>default value</span> provided by <code
+  data-x="xattr-ReflectDefault">[ReflectDefault]</code>.</p>
```
Provided by "the argument provided to" [ReflectDefault], or something like that?

domenic reviewed (approved):
This looks good to me and I'd be OK merging it as-is.

One more potential point for discussion: we could automatically lowercase the name, to remove many cases where you need to provide it. E.g. instead of `[Reflect="datetime"] attribute DOMString dateTime`, we could just do `[Reflect] attribute DOMString dateTime`.

I kind of like the current explicitness, but on the other hand maintaining such cases is just kind of busywork for everyone, and maybe we should make it automatic. Then the string-taking versions is reserved for truly-exceptional cases.

lukewarlow commented:
Ah yeah the lowercasing was something I'd thought of too forgot to comment about it, it seems webkits does that automatically at the very least. I think I lean towards doing it automatically just because it cuts down on noise and then the string case as you say can be for stuff where it's genuinely a different name (think encoding reflecting enctype or I think there's some hyphenated ones dotted about)

domenic commented:
Since this has no normative impact, there's no need for browser support, just spec editor support.

lukewarlow commented on source:
```diff
@@ -28992,13 +28973,13 @@ interface <dfn interface>HTMLPictureElement</dfn> : <span>HTMLElement</span> {
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>] attribute USVString <dfn attribute for="HTMLSourceElement" data-x="dom-source-srcset">srcset</dfn>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>] attribute DOMString <dfn attribute for="HTMLSourceElement" data-x="dom-source-sizes">sizes</dfn>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>] attribute DOMString <dfn attribute for="HTMLSourceElement" data-x="dom-source-media">media</dfn>;
+  [<span>CEReactions</span>, <span data-x="xattr-Reflect">Reflect</span>] attribute unsigned long <dfn attribute for="HTMLIFrameElement,HTMLEmbedElement,HTMLObjectElement,HTMLSourceElement,HTMLVideoElement" data-x="dom-dim-width">width</dfn>;
```
Yeah keeping the IDs was the main reason I did it this way.

domenic commented on source:
```diff
@@ -56732,40 +56447,26 @@ and a height of &lt;meter value=2>2cm&lt;/meter>.&lt;/p> &lt;!-- <strong>BAD!</s
-  setting, the given value must be converted to the <span>best representation of the number as a
-  floating-point number</span> and then the <code data-x="attr-meter-value">value</code> content
-  attribute must be set to that string.</p>
+  attribute, on getting, must return the <span data-x="concept-meter-actual">actual
```
I agree fixing these while we're here would be good, but I think if we do, we should also use `<span>this</span>'s`.

annevk commented on source:
```diff
@@ -55382,23 +55142,14 @@ interface <dfn interface>HTMLOptionElement</dfn> : <span>HTMLElement</span> {
   attribute, must return that attribute's value; otherwise, it must return the element's <span
-  data-x="concept-option-label">label</span>. On setting, the element's <code
-  data-x="attr-option-label">label</code> content attribute must be set to the new value.</p>
+  data-x="concept-option-label">label</span>.</p>
```
this + getter steps

lukewarlow commented on source:
```diff
@@ -57934,37 +57619,29 @@ form.method === input; // => true</code></pre>
+  attribute's getter must <span>reflect</span>, <span data-x="treated as a URL">as a URL</span>, the
+  <code data-x="attr-fs-action">action</code> content attribute, except that when the content
+  attribute is missing or its value is the empty string, the element's <span>node document</span>'s
+  <span data-x="concept-document-url">URL</span> must be returned instead. The <dfn attribute
```
I've udpated these not to use reflect. Hopefully the wording is right.

lukewarlow commented on source:
```diff
@@ -20207,10 +20241,9 @@ a friend lost to the
-    also used by the <code>q</code> element.</p>
+  [<span>CEReactions</span>, <span data-x="xattr-ReflectURL">ReflectURL</span>] attribute USVString <dfn attribute for="HTMLQuoteElement" data-x="dom-quote-cite">cite</dfn>;
+};</code></pre> <p class="note">The <code>HTMLQuoteElement</code> interface is also used by the
+<code>q</code> element.</p>
```
This is being caused by specfmt fwiw. I've reverted it but next time someone runs specfmt it'll make the change again.

lukewarlow commented:
Thanks for all the reviews!
